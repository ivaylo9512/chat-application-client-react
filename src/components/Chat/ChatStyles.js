import styled from 'styled-components';

export const ChatContainer = styled.div`
    height: 10.5vh;
    flex: 0 0 23vw;
    padding: 0 0.5%;
    background: white;
    justify-content: center;
    align-items: center;
    display: flex;
    margin-right: 0.5vw;
    cursor: pointer;
`
export const ChatNode = styled.div`
    padding: 0 5%;
    border-radius: 0.5vw;
    width: 95%;
    height: 75%;
    position: relative;
    background: rgb(24,27, 33);
    text-decoration: none;
    box-shadow: 0px 0px 0.15vw 0.15vw rgb(0 0 0);

    > div {
        height: 100%;
        width: 100%;
        align-items: center;
        display: flex;
    }
`
export const ImageContainer = styled.button`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 9.5vh;
    width: 9.5vh;
    border-radius: 50%;
    background: rgb(24,27, 33);
    box-shadow:0px 0px 0.2vw 0.09vw rgb(0, 0, 0, 0.8);

    img{
        border-radius: 50%;
        width: 5vh;
        height: 5vh;
        filter: drop-shadow(0px 1px 3px rgb(0 0 0));
    }
`

export const LastMsg = styled.span`
    color: #dbdada;
    text-decoration: overline;
    display: none;
    text-align: center;
`

export const Info = styled.div`
    flex: 1;
    font-size: 0.9vw;
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;

    b{
        color: white;
        vertical-align: bottom;
        text-align: center;
    }

    ${LastMsg} {
        display: ${props => props.displayInfo ? 'block' : 'none'};
      }
`

export const InfoButton = styled.button`
    height: 2.5vh;
    width: 2.5vh;
    font-size: 1.2vh;
    border-radius: 50%;
    cursor: pointer;
`