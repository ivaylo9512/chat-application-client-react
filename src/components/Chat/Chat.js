import { useState } from 'react';
import { useHistory } from 'react-router-dom'
import { useDispatch } from 'react-redux';
import { setCurrentChat } from 'app/slices/chatsSlice';
import { IMAGE_URL } from 'appConstants';
import { faInfo } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { ChatContainer, ChatNode, ImageContainer, Info, LastMsg, InfoButton } from './ChatStyles';

const Chat = ({chat, chat: {secondUser: { profileImage, firstName, lastName}}}) => {
    const dispatch = useDispatch();
    const history = useHistory();
    const [displayInfo, setDisplayInfo] = useState(false);
    
    const setChat = () => {
        dispatch(setCurrentChat(chat));
        history.push('/chat')
    }

    const toggleInfo = (e) => {
        e.stopPropagation();
        setDisplayInfo(!displayInfo)
    }

    return (
        <ChatContainer> 
            <ChatNode onClick={setChat}>
                <div>
                    <ImageContainer>
                        <img alt='profile image' src={`${IMAGE_URL}/${profileImage}`}/>
                    </ImageContainer>
                    <Info displayInfo={displayInfo}>
                        <b>{`${firstName} ${lastName}`}</b>
                        <LastMsg>{chat.lastMessage}</LastMsg>    
                    </Info>
                    <InfoButton onClick={toggleInfo}>
                        <FontAwesomeIcon icon={faInfo} />
                    </InfoButton>
                </div>
            </ChatNode>
        </ChatContainer>
    )
}

export default Chat