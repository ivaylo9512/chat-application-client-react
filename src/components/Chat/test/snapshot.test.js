import * as redux from 'react-redux';
import { shallow } from 'enzyme';
import Chat from 'components/Chat/Chat';

describe('Chat snapshot tests', () => {
    let wrapper;

    beforeEach(() => {
        jest.spyOn(redux, 'useDispatch').mockReturnValue(jest.fn());
    
        wrapper = shallow(<Chat chat={{id: 1, lastMessage: 'last message', secondUser: { firstName: 'first', lastName: 'last', profileImage: 'image1.png'} }}/>)
    })

    it('should rednder with snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

})