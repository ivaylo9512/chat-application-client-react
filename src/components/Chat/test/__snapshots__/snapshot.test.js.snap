// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat snapshot tests should rednder with snapshot 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    <div>
      <styled.button>
        <img
          alt="profile image"
          src="http://192.168.0.105:8083/api/files/download/image1.png"
        />
      </styled.button>
      <styled.div
        displayInfo={false}
      >
        <b>
          first last
        </b>
        <styled.span>
          last message
        </styled.span>
      </styled.div>
      <styled.button
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                192,
                512,
                Array [],
                "f129",
                "M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z",
              ],
              "iconName": "info",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </styled.button>
    </div>
  </styled.div>
</styled.div>
`;
